src/map/skill.cpp

```diff
@@ -24574,48 +24570,222 @@ static bool skill_parse_row_nocastdb(char* split[], int columns, int current)
 	return true;
 }
 
-/** Reads Produce db
- * Structure: ProduceItemID,ItemLV,RequireSkill,Requireskill_lv,MaterialID1,MaterialAmount1,...
- */
-static bool skill_parse_row_producedb(char* split[], int columns, int current)
-{
-	unsigned short x, y;
-	unsigned short id = atoi(split[0]);
-	t_itemid nameid = 0;
-	bool found = false;
+bool SkillProduceDatabase::add_itemconsumed(const ryml::NodeRef& node, std::shared_ptr<s_skill_produce_db_entry> &entry, bool isConsumed) {
```

Comment:
If you want, pass `entry` as `s_skill_produce_db_entry& entry`, then just use `entry.<blah>`

then call this function with `*entry`.